core/modules/layout/layout.context.admin.inc

```diff
@@ -0,0 +1,233 @@
+
+  $all_context_info = _layout_get_all_info('layout_context');
+  backdrop_sort($all_context_info, array('title'));
+  $form['context'] = array(
+    '#type' => 'select',
+    '#title' => t('Contexts'),
+    '#options' => array(),
+    '#required' => TRUE,
+    '#default_value' => isset($handler->plugin) ? $handler->plugin : NULL,
+    '#parents' => array('context'),
+    '#ajax' => array(
+      'wrapper' => 'context_settings',
+      'callback' => 'layout_context_ajax_style',
+    ),
+  );
+  foreach ($all_context_info as $key => $context_info) {
+    // @todo: add a "configurable" key to hook_context_info () instead?
+    if (!in_array($key, array('string', 'overrides_path'))) {
+      $form['context']['#options'][$context_info['name']] = $context_info['title'];
+    }
+  }
+
+  $form['context_settings'] = array(
+    '#type' => 'container',
+    '#id' => 'context_settings',
+    '#parents' => array('context_settings'),
+  );
+  $form['context_settings']['content'] = layout_context_return_form($form['context_settings'], $form_state);
+  $form['actions'] = array(
+    '#type' => 'actions',
+  );
+  $form['load_context_nojs'] = array(
+    '#type' => 'submit',
+    '#value' => t('Load context'),
+    '#submit' => array(
+      'layout_context_add_load_context_nojs',
+    ),
+    '#attributes' => array(
+      'class' => array('js-hide'),
+    ),
+  );
+  $form['actions']['add'] = array(
+    '#type' => 'submit',
+    '#value' => $is_new_context ? t('Add context') : t('Save context'),
+    '#submit' => array(
+      'layout_context_add_form_submit',
+    ),
+    '#validate' => array(
+      'layout_context_add_form_validate',
+    ),
+    '#ajax' => array(
+      'callback' => 'layout_ajax_form_save_dialog',
+    ),
+  );
+  $form['actions']['cancel'] = array(
+    '#type' => 'submit',
+    '#value' => t('Cancel'),
+    '#limit_validation_errors' => array(),
+    '#submit' => array(),
+    '#ajax' => array(
+      'callback' => 'layout_ajax_context_cancel_dialog',
+    ),
+  );
+  return $form;
+}
+
+/**
+ * AJAX callback to update the context settings.
+ */
+function layout_context_ajax_style($form, $form_state) {
+  return $form['context_settings'];
+}
+
+
+/**
+ * Submit handler to select a context. Hidden when JavaScript is enabled.
+ */
+function layout_context_add_load_context_nojs($form, &$form_state) {
+  $form_state['rebuild'] = TRUE;
+}
+
+/**
+ * Helper function to return a partial context settings form.
+ *
+ * This can be used to add a context to either a layout, block, or menu item,
+ * based on the values in the $form_state parameter.
+ */
+function layout_context_return_form($form, &$form_state) {
+  /* @var Layout $layout */
+  $layout = $form_state['layout'];
+  $context_id = $form_state['context_id'];
+
+  // If no context has yet been selected, there is no sub-form to display.
+  if (!isset($form_state['values']['context'])) {
+    return array();
+  }
+  else {
```

Comment:
Don't need an else here since if it gets this far it's always the case.

Reply:
done